src/actions/attack.cpp

```diff
@@ -200,8 +200,8 @@ battle_context_unit_stats::battle_context_unit_stats(nonempty_unit_const_ptr up,
 	damage_multiplier *= opp.damage_from(*weapon, !attacking, opp_loc, opp_weapon);
 
 	// Compute both the normal and slowed damage.
-	damage = round_damage(base_damage, damage_multiplier, 10000);
-	slow_damage = round_damage(base_damage, damage_multiplier, 20000);
+	damage = static_cast<int>(std::round(round_damage(base_damage, damage_multiplier, 10000)));
+	slow_damage = static_cast<int>(std::round(round_damage(base_damage, damage_multiplier, 20000)));
```

Comment:
What is the point of the cast and round? `round_damage` already returns an int?

Since base_damage is now a double `round_damage` needs to be adjusted to also take a double? Otherwise this still looks like truncation.

Reply:
satic_cast removed

Reply:
std::round removed